src/Avalonia.Controls/Window.cs

```diff
@@ -282,6 +284,32 @@ public Task<TResult> ShowDialog<TResult>()
                 return result.Task;
             }
         }
+        
+        private IDisposable GetModal()
+        {
+            var disabled = s_windows.Where(w => w.IsEnabled && w != this);
+            Window activated = null;
+            foreach (var window in disabled)
+            {
+                if (window.IsActive)
+                {
+                    activated = window;
+                }
+
+                window.IsEnabled = false;
+            }
+
+            PlatformImpl.Show();
```

Comment:
We should still be calling `PlatformImpl.ShowDialog` here as setting up a modal dialog may require special steps in the windowing subsystem.